wire-runtime/src/commonMain/kotlin/com/squareup/wire/internal/RuntimeMessageAdapter.kt

```diff
@@ -45,17 +45,32 @@ class RuntimeMessageAdapter<M : Any, B : Any>(
    * name.
    *
    * For reserved keywords in Kotlin or Java, a field like `public` are written as so although the
-   * generated field name is `public_`. We wanna read in either form. As well, in proto3 fields
+   * generated field name is `public_`. We want to read in either form. As well, in proto3 fields
    * declared in snake_case like `customer_id` are written in camelCase like `customerId`, but can
    * be read in either form. We can use exclusive logic between the two cases because there's no
    * keyword defined in snake_case. The alternate name will be absent if the field name isn't a
    * keyword or if the snake and camel cases are the same, such as in single-word identifiers.
+   * Lastly, in order to deserialize camelCased field from proto2, we add a camelCase alternative
```

Comment:
What's specific about proto2 here?

Since both versions are supposed to behave the same, is the proto2 comment effectively an implementation detail?

Reply:
Oh sorry, I think I follow now. Would it make sense to reword things in so far as the json_name value of the annotation is only provided for explicitly defined json names?

In other words, default/implicit value will not be present in the annotation. 

Or am I still misreading?

Reply:
Yes. Because proto2 has no spec around JSON mapping.